### PR TITLE
Emulation Data Extraction from xclbin

### PR DESCRIPTION
**Pull request by venkatp36**
Extracted the AIE shim emulation data from the xclbin and placed in the runs folder after unzip which will eventually be used by the Qemu

### CODE REVIEW

**Comment by sonals** [comment]
ok to test

File: src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 1]
```diff
@@ -253,13 +263,15 @@ namespace xclhwemhal2 {
     delete[] xmlFile;
     delete[] memTopology;
     delete[] pdi;
+    delete[] emuData;
 
     PRINTENDFUNC;
     return returnValue;
   }
```
Comment (sonals) [inline src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx]: 1. Please use early exits coding style; for example you can check for _xclbin_ magic number all in one place in the beginning and exit early if the magic number is incorrect which will remove the indentation
https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code
2. Drop all the legacy xclbin0/xclbin1
3. Use std::unique_ptr<char[]> for char buffer resource management. See dmatest.h for example.

File: src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 2]
```diff
@@ -571,9 +600,10 @@ namespace xclhwemhal2 {
           mLogStream << __func__ << " xocc command line: " << launcherArgs << std::endl;
 
         const char* simMode = NULL;
-        if ( pdi ) {
-          launcherArgs += " -pdi " + pdiFileName + " ";
+        if ( emuData ) {
```
Comment (sonals) [inline src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx]: No spaces around if condition.

File: src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.h [hunk 3]
```diff
@@ -118,7 +118,8 @@ using addr_type = uint64_t;
       int xclLoadXclBin(const xclBin *buffer);
       //int xclLoadBitstream(const char *fileName);
       int xclLoadBitstreamWorker(char* zipFile, size_t zipFileSize, char* xmlfile, size_t xmlFileSize,
-                                 char* debugFile, size_t debugFileSize, char* memTopology, size_t memTopologySize, char* pdi, size_t pdiSize);
+                                 char* debugFile, size_t debugFileSize, char* memTopology, size_t memTopologySize,
+                                 char* pdi, size_t pdiSize, char* emuData, size_t emuDataSize);
```
Comment (sonals) [inline src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.h]: Should the pointers be const?

File: src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx (hunk 1 above)
Comment (venkatp36) [reply src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx]: Hi Sonal,  updated as per the review comments.

File: src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx (hunk 2 above)
Comment (venkatp36) [reply src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.cxx]: updated as the per the review comments. Can you please push the changes. We are working on AIE design convergence. By Monday, we want all the changes to be in HEAD and need to work with bootgen team to take it further

**Comment by sonals** [comment]
retest this please

**Comment by venkatp36** [comment]
Looking at the build, seems some tests failed and those failures are not related to my pull request. Seems failures are due to some other pull requests. Is this being actively looked into?

**Comment by maxzhen** [comment]
retest this please

**Comment by maxzhen** [comment]
retest this please

File: src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.h (hunk 3 above)
Comment (venkatp36) [reply src/runtime_src/core/emulation/hw_em/generic_pcie_hal2/shim.h]: Hi Sonal, discussed with Vamsi and moved all the args to struct and pass that struct as an argument.

**Comment by venkatp-xilinx** [comment]
I could see some tests are failing with the Error:
**loading kernel.xclbin
./xsim.dir/emu_wrapper_behav/axsim: error while loading shared libraries: librdi_simulator_kernel.so: cannot open shared object file: No such file or directory**

This error is no way related to my PR. Seems one more infra issue

**Comment by maxzhen** [comment]
retest this please

**Comment by maxzhen** [comment]
Closing since it's already marked as invalid.

**Pull request closed without merging**